File: app/src/main/java/expressive/fan/core/MainActivityView.java
package expressive.fan.core;

public interface MainActivityView {
    void createPlayer();
    void play();
    void pause();
    void stop();

    void setPlayButtonEnabled(boolean b);
    void setPauseButtonEnabled(boolean b);
    void setStopButtonEnabled(boolean b);

    int getTotalAudioDuration();

    void seekTo(int newPos);

    void updateProgress(int position, int progress);

    int getCurrentPosition();
}

File: app/src/main/java/expressive/fan/core/MainActivityView.kt
package expressive.fan.core

interface MainActivityView {
    fun createPlayer()
    fun play()
    fun pause()
    fun stop()
    fun setPlayButtonEnabled(b: Boolean)
    fun setPauseButtonEnabled(b: Boolean)
    fun setStopButtonEnabled(b: Boolean)
    fun getTotalAudioDuration(): Int
    fun seekTo(newPos: Int)
    fun updateProgress(position: Int, progress: Int)
    fun getCurrentPosition(): Int
}